store/providers/xorm/cluster_types.go

```diff
@@ -24,37 +31,53 @@ func (Cluster) TableName() string {
 	return "ac_cluster"
 }
 
-func encodeCluster(in *api.Cluster) (*Cluster, error) {
-	cluster := &Cluster{
-		//Kind:              in.Kind,
-		//APIVersion:        in.APIVersion,
-		Name: in.Name,
-		//UID:               string(in.UID),
-		//ResourceVersion:   in.ResourceVersion,
-		//Generation:        in.Generation,
-		DeletedUnix: nil,
-	}
-	if in.DeletionTimestamp != nil {
-		cluster.DeletedUnix = types.Int64P(in.DeletionTimestamp.Time.Unix())
-	}
-	/*labels, err := json.Marshal(in.ObjectMeta.Labels)
+func EncodeCluster(in *api.Cluster) (*Cluster, error) {
+	ctx := context.Background()
+	secretId := stypes.RotateQuarterly()
+	urlstr := xkms.Scheme + "://" + secretId
+	keeper, err := secrets.OpenKeeper(ctx, urlstr)
 	if err != nil {
-		return nil, err
-	}*/
-	//cluster.Labels = string(labels)
+		return nil, fmt.Errorf("failed to open secret keeper: %v", err)
+	}
+	defer keeper.Close()
 
 	data, err := json.Marshal(in)
 	if err != nil {
 		return nil, err
 	}
-	cluster.Data = string(data)
+
+	cipher, err := keeper.Encrypt(ctx, data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encrypt: %v", err)
+	}
+
+	cluster := &Cluster{
+		Name:        in.Name,
+		Data:        cipher,
+		SecretID:    secretId,
+		DeletedUnix: nil,
+	}
+	if in.DeletionTimestamp != nil {
+		cluster.DeletedUnix = types.Int64P(in.DeletionTimestamp.Time.Unix())
+	}
 
 	return cluster, nil
 }
 
-func decodeCluster(in *Cluster) (*api.Cluster, error) {
+func DecodeCluster(in *Cluster) (*api.Cluster, error) {
+	ctx := context.Background()
```

Comment:
this block of code is same for `DecodeCertificate` method. move this  to a new function
```
	ctx := context.Background()
￼ 	urlstr := xkms.Scheme + "://" + in.SecretID
￼ 	keeper, err := secrets.OpenKeeper(ctx, urlstr)
￼ 	if err != nil {
￼ 		return nil, fmt.Errorf("failed to open secret keeper: %v", err)
￼ 	}
￼
￼ 	data, err := keeper.Decrypt(ctx, in.Data)
￼ 	if err != nil {
￼ 		return nil, fmt.Errorf("failed to decrypt: %v", err)
￼ 	}
```